File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/MangaPicture.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "manga_picture")
public class MangaPicture {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "mp_id")
    private Long mpId;

    @Lob
    @Column(name = "mp_picture", nullable = false)
    private byte[] mpPicture;

    @OneToOne
    @JoinColumn(name = "m_id", nullable = false, unique = true)
    private Manga manga;
}

File: Frontend/manga-world/src/Model/MangaPicture.ts
import Manga from "./Manga";

class mangaPicture {
  mpId: number;
  picture: string;
  manga: Manga;

  constructor(mpId: number, picture: string, manga: Manga) {
    this.mpId = mpId;
    this.picture = picture;
    this.manga = manga;
  }
}

export default mangaPicture;